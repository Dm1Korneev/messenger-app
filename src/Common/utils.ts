
type Filter = Parameters<typeof Array.prototype.filter>[0]
export const distinct: Filter = (value, index, self) => self.indexOf(value) === index;
