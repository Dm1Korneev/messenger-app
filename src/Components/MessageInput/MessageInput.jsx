import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

const MessageInput = ({ onSendMessage }) => {
  const [messageText, setMessageText] = useState('');

  const handleChange = (event) => {
    setMessageText(event.target.value);
  };

  const handleKeyPress = (event) => {
    if (event.key === 'Enter') {
      event.preventDefault();
      if (messageText.trim()) {
        onSendMessage({ messageText });
        setMessageText('');
      }
    }
  };

  return (
    <TextField
      value={messageText}
      onChange={handleChange}
      onKeyPress={handleKeyPress}
      multiline
      rows="2"
      rowsMax="5"
      fullWidth
      margin="normal"
      label="Write a message"
      variant="outlined"
    />
  );
};

MessageInput.propTypes = {
  onSendMessage: PropTypes.func.isRequired,
};

export default MessageInput;

