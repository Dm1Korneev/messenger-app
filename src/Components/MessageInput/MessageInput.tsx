import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import { useDispatch } from 'react-redux';

import { sendMessage } from 'Redux/actions';

export const MessageInput = () => {
  const dispatch = useDispatch();

  const [messageText, setMessageText] = useState('');

  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setMessageText(event.target.value);
  };

  const handleKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      event.preventDefault();
      if (messageText.trim()) {
        dispatch(sendMessage({ messageText }));
        setMessageText('');
      }
    }
  };

  return (
    <TextField
      value={messageText}
      onChange={handleChange}
      onKeyPress={handleKeyPress}
      multiline
      rows="2"
      rowsMax="5"
      fullWidth
      margin="normal"
      label="Write a message"
      variant="outlined"
    />
  );
};
