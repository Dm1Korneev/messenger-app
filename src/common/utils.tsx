export function distinct(value: any, index: number, self: Array<any>): boolean {
  return self.indexOf(value) === index;
}
