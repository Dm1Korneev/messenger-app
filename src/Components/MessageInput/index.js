import commonHoc from 'Components/commonHoc';

import { sendMessage } from 'Redux/actions';
import MessageInput from './MessageInput';

const mapDispatchToProps = {
  onSendMessage: sendMessage,
};

export default commonHoc(MessageInput, {
  mapDispatchToProps,
});

