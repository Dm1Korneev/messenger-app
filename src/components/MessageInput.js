import React, { Component } from "react";
import "./MessageInput.css";
import TextField from "@material-ui/core/TextField";

class MessageInput extends Component {
  constructor(props) {
    super(props);
    this.state = { messageText: "" };
  }

  handleChange = event => {
    this.setState({ messageText: event.target.value });
  };

  handleKeyPress = event => {
    if (event.key === "Enter" && this.state.messageText) {
      event.preventDefault();
      this.props.onSendMessage(this.state.messageText);
      this.setState({ messageText: "" });
    }
  };

  render() {
    const { messageText } = this.state;

    return (
      <div className="MessageInput">
        <TextField
          value={messageText}
          onChange={this.handleChange}
          onKeyPress={this.handleKeyPress}
          color="primary"
          multiline
          rows="2"
          rowsMax="5"
          fullWidth
          margin="normal"
          label="Write a message..."
          variant="outlined"
        />
      </div>
    );
  }
}

export default MessageInput;
