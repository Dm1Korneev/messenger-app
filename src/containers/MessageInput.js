import { connect } from 'react-redux';
import { sendMessage } from 'Redux/actions';
import MessageInput from 'Components/MessageInput';

const mapStateToProps = () => ({});

const mapDispatchToProps = (dispatch) => ({
  onSendMessage: (messageText) => dispatch(sendMessage(messageText)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MessageInput);
