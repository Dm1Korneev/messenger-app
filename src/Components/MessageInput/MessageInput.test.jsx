import React from 'react';
import { render } from '@testing-library/react';
import { mockComponent } from 'testing/utils';

import MessageInput from './MessageInput';

jest.mock('@material-ui/core/TextField', () => mockComponent('TextField'));

const props = {
  onSendMessage: jest.fn(),
};

describe('render MessageInput component', () => {
  const setup = () => render(
    <MessageInput {...props} />,
  );

  test('snapshot test', async () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});

