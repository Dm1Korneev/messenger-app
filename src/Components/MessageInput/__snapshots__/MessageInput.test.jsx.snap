// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MessageInput component snapshot test 1`] = `
<DocumentFragment>
  <div
    data-originalcomponent="TextField"
    test-prop-classname="\\"MessageInput-textField-3\\""
    test-prop-color="\\"primary\\""
    test-prop-fullwidth="true"
    test-prop-inputlabelprops="{\\"classes\\":{\\"root\\":\\"MessageInput-InputLabel-1\\"}}"
    test-prop-inputprops="{\\"classes\\":{\\"input\\":\\"MessageInput-Input-2\\"}}"
    test-prop-label="\\"Write a message...\\""
    test-prop-margin="\\"normal\\""
    test-prop-multiline="true"
    test-prop-rows="\\"2\\""
    test-prop-rowsmax="\\"5\\""
    test-prop-value="\\"\\""
    test-prop-variant="\\"outlined\\""
  />
</DocumentFragment>
`;
